test/conformance/context/urContextCreateWithNativeHandle.cpp

```diff
@@ -30,3 +30,66 @@ TEST_P(urContextCreateWithNativeHandleTest, Success) {
 
     ASSERT_SUCCESS(urContextRelease(ctx));
 }
+
+TEST_P(urContextCreateWithNativeHandleTest, SuccessWithOwnedNativeHandle) {
+    ur_native_handle_t native_context = nullptr;
+    if (urContextGetNativeHandle(context, &native_context)) {
```

Comment:
For this and the similar pattern in `DeviceCreateWithNativeHandle`, I think it would be better to make these ASSERT_SUCCESS instead, and update the pre-existing calls (i.e. [here](https://github.com/oneapi-src/unified-runtime/pull/1500/files#diff-948c9bd68a4224fc6d91be6b79fb7c00db0e01b180e6bebc96e7bb793f392217R13)) to the same. Looking at the match files it doesn't look like the `SKIP`/`continue` are doing anything to prevent fails, and we shouldn't really allow this call to fail anyway

Reply:
Fixed

Reply:
alright looks like
> Looking at the match files it doesn't look like the SKIP/continue are doing anything to prevent fails

is not true for pretty much every object other than context. I was going to write an issue about documenting this but we actually do document returning `UR_RESULT_ERROR_UNSUPPORTED_FEATURE` "If the adapter has no underlying equivalent handle. " (you can see that under any native handle entry point). I think [urContextGetNativeHandle](https://github.com/oneapi-src/unified-runtime/blob/main/test/conformance/context/urContextGetNativeHandle.cpp#L14) gets it right, with that pattern we at least make sure it's the error code we're expecting.

Reply:
Yeah, good point, I didn't notice that in the spec. I changed it and covered `CreateWithNativeHandle` with that too as it also have the same possibility described in the spec.

Reply:
cool thanks, are the match files still accurate? seems like [this](https://github.com/oneapi-src/unified-runtime/blob/main/source/adapters/cuda/context.cpp#L155) should skip now but it's still here https://github.com/oneapi-src/unified-runtime/pull/1500/files#diff-27bfe80115e9399ca467c92d4bb10e9b5687728cf94bdbed791016aa0e0855d7L1

Reply:
It is a deletion change as this file have only this test that was failing in the past, I have updated it to delete this line as it should skip now.

Reply:
yeah sorry I was basically asking if you had updated the match files now that most of the native handle tests should be skipping for cuda and hip